ocaml/xapi/repository_helpers.ml

```diff
@@ -1478,3 +1524,114 @@ let set_pending_guidances ~ops ~coming =
            to_be_removed ;
          do_with_vm_pending_guidances ~op:ops.vm_add ~vm:vm_ref_str to_be_added
      )
+
+let failure_of_livepatch_component = function
+  | Livepatch.Xen ->
+      Guidance.RebootHostOnXenLivePatchFailure
+  | Livepatch.Kernel ->
+      Guidance.RebootHostOnKernelLivePatchFailure
+
+let component_of_livepatch_failure = function
+  | Guidance.RebootHostOnXenLivePatchFailure ->
+      Some Livepatch.Xen
+  | Guidance.RebootHostOnKernelLivePatchFailure ->
+      Some Livepatch.Kernel
+  | _ ->
+      None
+
+let merge_livepatch_failures ~previous_failures ~applied ~failed =
```

Comment:
How about rename to `merge_livepatch_failure_guidances`?

Reply:
I think it's clear in this context (in function update_livepatch_failures).